projects/docs/docs/commands/graph.md

```diff
@@ -52,9 +56,84 @@ If you prefer the old style, without different colors and shapes, pass the `--si
+| ---------- | ----------------------------------------------------- |
+| `name`     | Name of the manifest.                                 |
+| `path`     | Absolute path to the manifest.                        |
+| `projects` | A map of project absolute paths to `Project` objects. |
+
+##### Project
+
+| Key        | Description                   |
+| ---------- | ----------------------------- |
+| `name`     | Name of the project.          |
+| `path`     | Absolute path to the project. |
+| `targets`  | A list of `Target` objects.   |
+| `schemes`  | A list of `Scheme` objects.   |
+| `packages` | A list of `Package` objects.  |
+
+##### Target
+
+| Key       | Description                                                                                                             |
+| --------- | ----------------------------------------------------------------------------------------------------------------------- |
+| `name`    | Name of the target.                                                                                                     |
+| `product` | The product type of the target. See (Product)[https://docs.tuist.io/manifests/project#product] for all possible values. |
+
+##### Scheme
+
+| Key                 | Description                                                     |
+| ------------------- | --------------------------------------------------------------- |
+| `name`              | Name of the scheme.                                             |
+| `testActionTargets` | A list of targets names that are tested as part of this scheme. |
+
+##### Package
+
+| Key    | Description                                                                                                      |
+| ------ | ---------------------------------------------------------------------------------------------------------------- |
+| `kind` | The type of the package. See (Package)[https://docs.tuist.io/manifests/project#package] for all possible values. |
+| `path` | Absolute path to the package.                                                                                    |
+
+Sample JSON output:
+```
+{
+    "name": "NAME_OF_MANIFEST",
+    "path": "PATH_TO_MANIFEST_DIR",
+    "projects": {
+        "PATH_TO_PROJECT": {
+            "name": "NAME_OF_PROJECT",
+            "path": "PATH_TO_PROJECT",
+            "targets": [
+                {
+                    "name": "NAME_OF_TARGET",
+                    "product": "TYPE_OF_PRODUCT"
+                }
+            ],
+            "schemes": [
+                {
+                    "name": "NAME_OF_SCHEME",
+                    "testActionTargets": [
+                        "TEST_ACTION_TARGET"
+                    ]
+                }
+            ],
+            "packages": [
+                {
+                    "kind": "KIND_OF_PACKAGE_REMOTE_OR_LOCAL",
+                    "path": "PATH_OR_URL_OF_PACKAGE"
+                }
+            ]
+        }
+    }
+}
+```
```

Comment:
Are all the paths relative? I would like the generated graph to be the same regardless of the manifest path (hence I would probably remove the top-level path if not needed)

Reply:
> Are all the paths relative? I would like the generated graph to be the same regardless of the manifest path (hence I would probably remove the top-level path if not needed)

I am not sure if I agree with this. If e.g. all source paths are relative, it will actually make it harder to work with.

Reply:
Why would it make it harder to work it? if they a relative to the path when the command is invoked I don't see a problem

Reply:
The scenario I am thinking of is going through `sources` to e.g. lint all the files. Also note we are working with strings here and not `RelativePath`, making the conversion a little bit harder.

What would be a good use-case for having relative paths here?

Reply:
Going from relative to absolute should be just a matter of prepending a fixed path, so I don't think it's a big issue.
Any logic you would do on the files, I expect you do it from the project root, so you should be able to use relative paths directly.

I don't have a strong use-case for relative paths, but even being able to compare generated JSONs on different machines could be a nice to have.
I think we should go for relative by default everywhere, unless we have a strong reason to use absolute.

Makes sense? 🚀

Reply:
I'm fine with that, once we start using this in tasks, we should provide a utility that will convert a string of relative path to a string of absolute path, so it's as easy-as-possible for users to use.

Reply:
For our use case we actually must have absolute paths. Using relative paths would make our life a lot more painful. Using absolute paths is also consistent with SPM's output. I feel pretty strongly that we should stick with absolute paths here.

Reply:
Fair! The current format is labeled as unstable, so if we find relative paths would actually be better, I'd revert it. But so far it seems that absolute paths are a better option here.